app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,45 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def find_host_by_provider_id(provider_id)
+      providers = {
+        "gce" => [ManageIQ::Providers::Google::CloudManager, :name],
+        "aws" => [ManageIQ::Providers::Amazon::CloudManager, :uid_ems],
+      }
+      scheme, instance_uri = provider_id.split("://", 2)
+      prov, name_field = providers[scheme]
+      return nil unless prov
```

Comment:
Taste: @enoodle personally I find this `return` here a little bit ugly. And I'd go with a little bit more validation:

```
instance_id = instance_uri.split('/').last
prov::Vm.find_by({name_field => instance_id}) if !prov.nil? and !instance_id.is_blank?
```